### PR TITLE
BUG: Ignore UTF-8 decode errors

### PR DESCRIPTION
**Pull request by talibhmukadam**
> Problem
Some pdfs contain Latin characters, and when trying to read them using pypdf, it throws the following exception.|

```Traceback (most recent call last):
    text = page.extract_text()
  File "/Users/.pyenv/versions/3.10.3/lib/python3.10/site-packages/pypdf/_page.py", line 1851, in extract_text
    return self._extract_text(
  File "/Users/.pyenv/versions/3.10.3/lib/python3.10/site-packages/pypdf/_page.py", line 1356, in _extract_text
    content = ContentStream(content, pdf, "bytes")
  File "/Users/.pyenv/versions/3.10.3/lib/python3.10/site-packages/pypdf/generic/_data_structures.py", line 877, in __init__
    self.__parse_content_stream(stream_bytes)
  File "/Users/.pyenv/versions/3.10.3/lib/python3.10/site-packages/pypdf/generic/_data_structures.py", line 943, in __parse_content_stream
    operands.append(read_object(stream, None, self.forced_encoding))
  File "/Users/.pyenv/versions/3.10.3/lib/python3.10/site-packages/pypdf/generic/_data_structures.py", line 1053, in read_object
    return read_string_from_stream(stream, forced_encoding)
  File "/Users/.pyenv/versions/3.10.3/lib/python3.10/site-packages/pypdf/generic/_utils.py", line 107, in read_string_from_stream
    msg = rf"Unexpected escaped string: {tok.decode('utf8')}"
    ```

### CODE REVIEW

**Review by tasfiqul-ghani — APPROVED** [review]
Facing this issue :  'utf-8' codec can't decode byte 0x83 in position 0: invalid start byte
This PR will fix the issue.

**Comment by tasfiqul-ghani** [comment]
@MartinThoma  Please merge the PR.It will fix a major issue. For some PDFs we are getting this error :
 'utf-8' codec can't decode byte 0x83 in position 0: invalid start byte

**Comment by pubpub-zz** [comment]
@tasfiqul-ghani 
can you share one pdf showing the issue ?

File: pypdf/generic/_utils.py [hunk 1]
```diff
@@ -104,7 +104,7 @@ def read_string_from_stream(
                     # line break was escaped:
                     tok = b""
                 else:
-                    msg = rf"Unexpected escaped string: {tok.decode('utf8')}"
+                    msg = rf"Unexpected escaped string: {tok.decode('utf8','ignore')}"
```
Comment (pubpub-zz) [inline pypdf/generic/_utils.py:107]:
```suggestion
                    msg = f"Unexpected escaped string: {tok.decode('utf-8','ignore')}"
```
this is would be more consistent with the rest of the code

**Review by pubpub-zz — APPROVED** [review]

**Comment by MartinThoma** [comment]
Thank you for your contribution @talibhmukadam :pray: 

If you want, I can add you to https://pypdf.readthedocs.io/en/latest/meta/CONTRIBUTORS.html :-)

**Comment by pubpub-zz** [comment]
@talibhmukadam / @tasfiqul-ghani 
I' m still understanding in a sample file. I would like to check if nothing is hidden behind🤔

**Comment by talibhmukadam** [comment]
@MartinThoma , @pubpub-zz thank you guys for reviewing and merging the fix so fast. If I could ask, what would be a possible ETA to release the new version of pypdf with this fix? I am sorry to rush you but it is blocking us from releasing a new feature 😢 

@MartinThoma , Yes, please feel free to add me to the contributor's list. 😄 

@pubpub-zz , I would love to share the pdf file, but the few pdf files that we got the errors on, are resume files that contain PII information which my organization wouldn't allow me to share. I hope you understand. 
If I could reproduce this error on any other pdf file. I will definitely share that with you.

**Comment by MartinThoma** [comment]
I'm creating a release at the moment. It will be on PyPI in less than 2 hours.

However, if you want to make sure that the fix stays in pypdf, we need to get a sample file. Otherwise it could happen in future that another change breaks it again (but I also understand the PII restrictions :cry: )

**Comment by MartinThoma** [comment]
> Yes, please feel free to add me to the contributor's list. smile

Which name should I use and should I link to some profile (e.g. your Github profile?)

**Comment by pubpub-zz** [comment]
@talibhmukadam / @tasfiqul-ghani 
Would you agree to share private the file. if so email it to @MartinThoma  info@martin-thoma.de

**Pull request merged**